### PR TITLE
Spotify Theme improvements

### PR DESCRIPTION
**Pull request by Rishabh-malhotraa**
# Improvement to the Spotify theme

Screenshot from 2021-04-07 20-13-44

Basing on the Spotify web player I made some improvements to the Spotify theme

1. Made the search bar like that on Spotify
2. Changed the scrollbar to look more like how it is on spotify

|  New-Changes  | Old |
| ------------- | ------------- |
| Screenshot from 2021-04-07 20-09-16 | Screenshot from 2021-04-07 20-16-11 
|Screenshot from 2021-04-07 20-20-11|Screenshot from 2021-04-07 20-18-33|

### CODE REVIEW

**Comment by deluan** [comment]
I'd like to have a review from @samarsault, who created this Spotify-ish theme

File: ui/src/themes/spotify.js [hunk 1]
```diff
@@ -337,6 +365,20 @@ export default {
...
+            '*::-webkit-scrollbar-track': {
+              background: 'rgb(15, 15, 15)',
+            },
+            '*::-webkit-scrollbar-thumb': {
+              backgroundColor: '#555',
+              '&:hover': {
+                background: '#898989',
+              },
+            },
+          },
```
Comment (samarsault) [inline ui/src/themes/spotify.js:381]: Why do we need an additional nesting of `@global` here?

**Comment by samarsault** [comment]
The styles for the search seem to break the filters (also in RaFilter->form)

Screenshot 2021-04-07 at 10 09 16 PM

**Comment by deluan** [comment]
I actually think that the "Search" bar in this PR is a good fit for the Omni search bar we want to introduce (similar to Spotify), but not to be used in the current search fields. They are more like Filters than Search bars. Maybe we can do this now, but will have to revert when the Omni bar is introduced

File: ui/src/themes/spotify.js (hunk 1 above)
Comment (Rishabh-malhotraa) [reply ui/src/themes/spotify.js:381]: I am not sure why it does not work without the extra scoring I basically took help of stack overflow when I was unable to get it to work, and there it mentioned to score with global again, I feel it has something to do with the specificity of applying reset-styles (::webkit- ::before etc..)

**Comment by Rishabh-malhotraa** [comment]
> The styles for the search seem to break the filters (also in RaFilter->form)
> 
> Screenshot 2021-04-07 at 10 09 16 PM

@samarsault Nice catch! I did not find that when I was testing thank you, it seems fixed now but do let me know if you find any  other bug too :P

File: ui/src/themes/spotify.js [hunk 2]
```diff
@@ -311,6 +311,38 @@ export default {
...
+        },
+      },
+    },
+    RaFilter: {
+      form: {
+        '& .MuiOutlinedInput-input:-webkit-autofill': {
+          '-webkit-box-shadow': '0 0 0 100px #28282b inset',
+          '-webkit-text-fill-color': 'white',
+        },
```
Comment (Rishabh-malhotraa) [inline ui/src/themes/spotify.js:345]: This gets rid of the obnoxious blue color we got when we autocomplete in filtered fields

Screenshot from 2021-04-07 20-18-33

**Comment by samarsault** [comment]
Search and other fields look good now. I can't test the scrollbar on my system i.e. macOS, since it auto hides/shows as required.

**Comment by Rishabh-malhotraa** [comment]
@deluan Reminder :P

**Comment by deluan** [comment]
Merging it, but without the Scrollbar styling. 

Maybe we should simulate macOS's scrollbar behaviour, with something like this: https://gist.github.com/spemer/a0e218bbb45433bd611e68446523a00b

**Comment by deluan** [comment]
Merged in https://github.com/navidrome/navidrome/pull/2012

**Pull request closed without merging**